lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/LineDeserializerUtils.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.line;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.hive.formats.HiveFormatUtils;
+import io.trino.hive.formats.line.openxjson.JsonString;
+import io.trino.plugin.base.type.DecodedTimestamp;
+import io.trino.plugin.base.type.TrinoTimestampEncoder;
+import io.trino.spi.TrinoException;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.AbstractType;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.VarcharType;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.hive.formats.line.openxjson.JsonWriter.canonicalizeJsonString;
+import static io.trino.hive.formats.line.openxjson.JsonWriter.writeJsonArray;
+import static io.trino.hive.formats.line.openxjson.JsonWriter.writeJsonObject;
+import static io.trino.spi.StandardErrorCode.BAD_DATA;
+import static io.trino.spi.type.Chars.truncateToLengthAndTrimSpaces;
+import static io.trino.spi.type.Varchars.truncateToLength;
+
+public final class LineDeserializerUtils
+{
+    private LineDeserializerUtils() {}
+
+    public static RuntimeException parseError(String message)
+    {
+        return parseError(message, Optional.empty());
+    }
+
+    public static RuntimeException parseError(String message, Optional<Throwable> throwable)
+    {
+        String errorPrefix = "Error Parsing a column in the table: ";
+        return throwable.map(e -> new TrinoException(BAD_DATA, errorPrefix + message, e)).orElseGet(() -> new TrinoException(BAD_DATA, errorPrefix + message));
+    }
+
+    public static void writeLong(AbstractType type, BlockBuilder builder, long value)
+    {
+        try {
+            type.writeLong(builder, value);
```

Comment:
This will result in a metamorphic call site and may have performance implications.